fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/SumoAmbassador.java

```diff
@@ -306,16 +306,25 @@ private void addNotYetAddedVehicles(long time) throws InternalFederateException
             int departIndex = vehicleRegistration.getDeparture().getDepartureConnectionIndex();
             String departSpeed = extractDepartureSpeed(vehicleRegistration);
             String laneId = extractDepartureLane(vehicleRegistration);
+            if (externalVehicleMap.containsKey(vehicleId)) {
+                if (externalVehicleMap.get(vehicleId).isAdded()) {
+                    iterator.remove();
+                    continue;
+                }
```

Comment:
In each iteration of this loop you currently do up to 4 lookups in `externalVehicleMap`. Better do
```java
ExternalVehicleState externalVehicleState = externalVehicleMap.get(vehicleId);
if (externalVehicleState  != null) {
  if (externalVehicleState.isAdded()){
  ...
  }
...
}

...

if (externalVehicleState != null) {
  externalVehicleState.setAdded(true);
}
```